src/test/ui/attr-eq-token-tree.rs

```diff
@@ -1,6 +1,4 @@
-// compile-pass
+#![feature(custom_attribute)]
 
-#![feature(custom_attribute, unrestricted_attribute_tokens)]
-
-#[my_attr = !] // OK under feature gate
+#[my_attr = !] //~ ERROR unexpected token: `!`
```

Comment:
Could you split this into a new separate feature gate instead so that folks can continue their experimentation with this on nightly? suggestion: `unrestricted_kv_attribute_tokens`

Reply:
I'm skeptical about this.
The feature was available for [almost two years](https://github.com/rust-lang/rust/pull/40346), but crater still found zero uses of it.

The question about key-value attributes if "what subset of `#[attr(stuff)]` we may want to write as `#[attr = stuff]`", so getting more experience with `#[attr(stuff)]` seems enough.

Reply:
The most requested extension that I've seen is supporting macro calls like `#[attr = mac!(args)]`, but it 1) doesn't fit into the current implementation, and most importantly 2) needs eager expansion.

Reply:
What I've seen the most of in the wild is `#[attr = "path::to::thing"]`, which could be better written as `#[attr = path::to::thing]`, but that also doesn't fit into the current "`unrestricted_kv_attribute_tokens`".

Reply:
Fair enough; let's come back with an RFC for the rest before doing it in nightly.

However, I'd like to note that this would be useful for things like `#[proptest(strategy = 0..10)]`, `#[serde(bound = T: MyTrait)]`, `#[serde(borrow = 'a + 'b)]` or other attributes with rich customization. Today, these usually wrap expressions, bounds, types, etc. in strings. I think the reason no one used the nightly thing is that it's not the worth the price of nightly when a string works. But it would be nice to not have to wrap in a string.

All in all, I'm quite interested in extending the grammar eventually.